### PR TITLE
Switch to SHA256 for detecting changes to files

### LINKED ISSUE #358: MD5 as a stamp algo
I'm new to doorstop and find it very interesting. I'm a bit concerned about the fact that fingerprinting as per Stamp class implementation here:

https://github.com/jacebrowning/doorstop/blob/5f624c80f81c950fedc30af34852979b67d57597/doorstop/core/types.py#L508) 

is still using MD5 hashing algorithm. Apart from being deprecated since long ago for any cryptography use (which is not the case here) it's also something not easily found on any modern application. Git itself is using SHA1 for creating hashes, and I suspect any expert body reviewing doorstop would regard using MD5 as suspicious as a minimum. Collision is not far from happening, and mangling for causing collision is also possible. Before the develpment goes too far would it be a good idea to (1) switch to SHA1 and (2) include a reference to the used algorithm in each hash so that any future change would not break compatibility?

### PR DESCRIPTION
**Pull request by jacebrowning**
Resolves #358 

Note that this is a breaking change so it will require a bump in the major version number to be released.

### CODE REVIEW

File: doorstop/core/types.py [hunk 1]
```diff
@@ -558,12 +559,12 @@ def yaml(self):
...
-    def digest(*values):
+    def digest(*values) -> str:
         """Hash the values for later comparison."""
-        md5 = hashlib.md5()
+        sha = hashlib.sha256()
         for value in values:
-            md5.update(str(value).encode())
-        return md5.hexdigest()
+            sha.update(str(value).encode())
+        return b85encode(sha.digest()).decode('utf-8')
```
Comment (jacebrowning) [inline doorstop/core/types.py]: Base85 is more concise, but still safely storable in YAML files.

**Comment by sebhub** [comment]
Thanks for changing the hash to SHA256.

I would use base64 encoding instead of base85 since this makes it a bit easier/safer for other consumers to use the value as is, e.g. in reports. It is just four bytes more compared to base85. My choice would be urlsafe_b64encode(). You can directly grep this value for example.

**Comment by sebhub** [comment]
Is it really safe to use base85 with YAML. I think something like this
```yaml
reviewed: !!foobar...
```
will be interpreted as a tag leading to a "could not determine a constructor for the tag" YAML parser error.

**Comment by sebhub** [comment]
I would like to start using doorstop for some real work and the secure hash is important for me. I would like to add a new pull request which

1. changes the hash algorithm to SHA256, and

2. uses URL-safe base64 encoding.

**Comment by jacebrowning** [comment]
@sebhub Sounds good! I'll close this one.

**Pull request closed without merging**